### PR TITLE
Let Travis send email notifications for build statuses

### PR DESCRIPTION
**Pull request by shindere**
The notifications are sent to the dedicated OCaml CI notifications list.

At the moment, the default configuration of both Travis and
AppVeyor is used to decide when to send out notifications.
This can always be fine-tuned later, if desired.

### CODE REVIEW

**Review by dra27 — APPROVED** [review]
LGTM - I have similar a set-up on my personal AppVeyor/Travis fork

**Comment by shindere** [comment]
David Allsopp (2017/08/31 00:17 -0700):
> dra27 approved this pull request.
> 
> LGTM - I have similar a set-up on my personal AppVeyor/Travis fork

Thanks a lot, @dra27. Is your setting on your personal repo enough for
you to get build status notifications for your PRs?

**Comment by dra27** [comment]
I think I have it set so I only get failure/change notification on Travis but everything on AppVeyor (or maybe the other way around!) - but I get a steady stream of emails when I'm working on a branch and pushing it... now I often have branch `foo` which is my pull request and `foo-dev` which contains test commits so that my Travis/AppVeyor confirms they work before 'hogging' the OCaml ones (for AppVeyor on your own fork, it's at the moment necessary to edit `appveyor.yml`, but avsm and I will be working on this after ICFP - see for example, https://github.com/dra27/ocaml/commits/unsigned-value-dev)

**Comment by shindere** [comment]
David Allsopp (2017/08/31 00:32 -0700):
> I think I have it set so I only get failure/change notification on
> Travis but everything on AppVeyor (or maybe the other way around!) -
> but I get a steady stream of emails when I'm working on a branch and
> pushing it...

So what I understand is that, with this seting, you are able to test
your branches even you create PRs for them, is that correct?

> now I often have branch `foo` which is my pull request and `foo-dev`
> which contains test commits so that my Travis/AppVeyor confirms they
> work before 'hogging' the OCaml ones (for AppVeyor on your own fork,
> it's at the moment necessary to edit `appveyor.yml`, but avsm and I
> will be working on this after ICFP - see for example,
> https://github.com/dra27/ocaml/commits/unsigned-value-dev)

What do you plan to do, more precisely?

From what I have read on Travis' website, the way it tests a PR is by
not testing the PR branch directly, but rather by testing the merge of
that branch into the latest upstream trunk. Due to this, I am a bit
worried that the present PR could create problems for you since the
merge may create conflicts on the .travis.yml file. What do you think?

**Comment by dra27** [comment]
Actually, that's a good point - by placing the email address in .travis.yml and appveyor.yml, that means that any user (like me) who also has Travis and AppVeyor set-up on their ocaml fork will start trying to email ocaml-ci-notifications.

At the moment, because appveyor.yml has a `branches` section, it doesn't test my personal branches until they are turned into pull requests, hence temporary commits like https://github.com/dra27/ocaml/commit/3a36eb8f4e283bddc7d487966b629fb764f961a4 - but the change we can make is that Anil can configure OCaml's AppVeyor to have those branches listed in the web interface instead (so ocaml/ocaml would only test trunk, 4.04 and 4.05 but individual users' forks would test whatever they configured).

I think maybe both these changes need to be made in the web interface - pinging @avsm for AppVeyor. Not sure what to do about Travis, as you can only configure its stuff in the .travis.yml.

**Comment by dra27** [comment]
Ah, I think Travis should work as expected - https://docs.travis-ci.com/user/notifications/#How-is-the-build-email-receiver-determined%3F suggests that most user's fork should not send email notifications when its on their own fork. Perhaps we should just experiment - we can always turn it off if we get too many emails.

So, I think the .travis.yml change is fine to test but the appveyor.yml change should be made by Anil using the web interface instead?

**Comment by shindere** [comment]
David Allsopp (2017/08/31 07:59 +0000):
> Ah, I think Travis should work as expected -
> https://docs.travis-ci.com/user/notifications/#How-is-the-build-email-receiver-determined%3F suggests that most user's fork should not send email notifications when its on their own fork. Perhaps we should just experiment - we can always turn it off if we get too many emails.

Agreed.

> So, I think the .travis.yml change is fine to test but the
> appveyor.yml change should be made by Anil using the web interface
> instead?

I see two options, here.

(1) We wait on his feedback and if that's okay we undo the change.

(2) I change the PR to just modify Travis' configuration, proceed with
merging and let AppVeyor for later.

Any preference?

**Comment by damiendoligez** [comment]
cc @avsm, as I believe you guys haven't properly mentioned his handle yet.

**Comment by shindere** [comment]
Damien Doligez (2017/09/01 14:07 +0000):
> cc @avsm, as I believe you guys haven't properly mentioned his handle
> yet.

You are correct, sorry about that! Thanks!

**Comment by avsm** [comment]
Sorry, I completely missed this in the ICFP rush! I have made the relevant changes in Appveyor so the email notifications go to the inria email address. I will also migrate our Appveyor setup to an organisation next week, so that the same people that have access to GitHub will have access to the CI config.

**Comment by shindere** [comment]
Anil Madhavapeddy (2017/09/15 12:45 +0000):
> Sorry, I completely missed this in the ICFP rush! I have made the
> relevant changes in Appveyor so the email notifications go to the
> inria email address. I will also migrate our Appveyor setup to an
> organisation next week, so that the same people that have access to
> GitHub will have access to the CI config.

Great, thanks! So am I correct that this PR needs to be updated to
include only the Travis change?

**Comment by avsm** [comment]
@shindere yes that's right. There's no way to configure Travis via the web interface for email notifications. However, Travis only sends emails to people with push access, and so most forks will not trigger emails in our case.

**Comment by shindere** [comment]
Anil Madhavapeddy (2017/09/15 12:54 +0000):
> @shindere yes that's right. There's no way to configure Travis via the
> web interface for email notifications. However, Travis only sends
> emails to people with push access, and so most forks will not trigger
> emails in our case.

Thanks, Anil. PR rebased and updated. Should be ready to be merged,
hopefully.

**Comment by mshinwell** [comment]
This seems fine.

**Pull request merged**
